### PR TITLE
LPS-145480 Test for dynamic collection with combined segments

### PR DESCRIPTION
**Pull request by YurenaC**
**Motivation**
=======

Impedibug test for issue: Dynamic collections are not properly working with combined segments in content display

**Proposed Solution**
=======

Review how the content display recovers all the items to be show related to the user and segments where they belongs

**Steps to Verify:**
=======

1. Go to Site Builder > Collections
2. Create a Dynamic Collection
3. Create some variations with different content and for the previous segments
4. Create a content page with Asset Publisher and another with Collection Display widgets
5. Go to System Settings > Assets > Collections
6. Enable the combine segments option for Dynamic Collections
7. Go to the content pages
8. Only the variation's content where the user belongs is shown. Also, the content only appear once.

**Change summary:**
=======

LPS-145480 Test dynamic collection combined assets shown

### CODE REVIEW

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+import org.junit.runner.RunWith;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Yurena Cabrera
+ */
+@RunWith(Arquillian.class)
+public class AssetListAssetEntryProviderImplTest {
```
Comment (cgoncas) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java:76]: I think this test should be under com.liferay.asset.list.asset.entry.provider.test.AssetListAssetEntryProviderTest, as we always test the API, no the Impl

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+			String.format("(firstName eq '%s')", user.getFirstName() + "a"),
+			Criteria.Conjunction.AND);
+
+		return SegmentsTestUtil.addSegmentsEntry(
+			groupId, CriteriaSerializer.serialize(criteria),
+			User.class.getName());
+	}
+
+	private String _getTypeSettings() {
+		return UnicodePropertiesBuilder.create(
```
Comment (cgoncas) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java:206]: Do you need we need all this properties?

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java (hunk 1 above)
Comment (YurenaC) [reply modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java:76]: I'll move it there, thanks!

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java (hunk 2 above)
Comment (YurenaC) [reply modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java:206]: Refactored!

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java [hunk 3]
```diff
@@ -73,6 +107,69 @@ public void setUp() throws Exception {
...
+		Dictionary<String, Object> dictionary =
+			HashMapDictionaryBuilder.<String, Object>put(
+				"enabled", true
+			).build();
+
+		try (ConfigurationTemporarySwapper configurationTemporarySwapper =
+				new ConfigurationTemporarySwapper(
+					"com.liferay.asset.list.web.internal.configuration." +
+						"FFCollectionsVariationsPrioritizationConfiguration",
+					dictionary)) {
```
Comment (cgoncas) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java:121]: As dictionary is only used here, maybe you can use the HashMapDictionaryBuilder directly

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java [hunk 4]
```diff
@@ -73,6 +107,69 @@ public void setUp() throws Exception {
...
+
+			SegmentsEntry segmentsEntry2 = _addSegmentsEntry(
+				_group.getGroupId(), TestPropsValues.getUser());
+
+			long[] segmentsEntryIds = {
+				segmentsEntry1.getSegmentsEntryId(),
+				segmentsEntry2.getSegmentsEntryId()
+			};
+
+			AssetListEntrySegmentsEntryRel variation1 =
```
Comment (cgoncas) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java:140]: Following Liferay's style book, this variable should be called assetListEntrySegmentsEntryRel1

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java [hunk 5]
```diff
@@ -73,6 +107,69 @@ public void setUp() throws Exception {
...
+				AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
+					_group.getGroupId(), assetListEntry,
+					segmentsEntry1.getSegmentsEntryId());
+
+			AssetListEntrySegmentsEntryRel variation2 =
+				AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
```
Comment (cgoncas) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java:146]: Following Liferay's style book, this variable should be called assetListEntrySegmentsEntryRel2

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java [hunk 6]
```diff
@@ -73,6 +107,69 @@ public void setUp() throws Exception {
...
+					_group.getGroupId(), assetListEntry,
+					segmentsEntry2.getSegmentsEntryId());
+
+			variation1.setPriority(1);
```
Comment (cgoncas) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java:150]: What do you think about create a new method addAssetListEntrySegmentsEntryRel with the priority, so we don't need to mutate that variable?

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java (hunk 4 above)
Comment (YurenaC) [reply modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java:140]: I thought about this but, that name does not tell me what I am really doing don't you think? If I tell you that it is a variation isn't it more clear for you that have to read the test now?

Happy to change if it is a style that needs to be enforced but I think that the readability of the test is also important

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java (hunk 3 above)
Comment (YurenaC) [reply modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java:121]: Done

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:forward

**Comment by YurenaC** [comment]
ci:forward

**Pull request closed without merging**